File: src/apple/Packages/MapleUI/Sources/MapleUI/Internal/MuiSplitLayoutMath.swift
// MuiSplitLayoutMath.swift — pure width-clamping and collapse-decision
// math for MuiSplitLayout (Templates, unified-component-catalog.md §5).
// Kept side-effect-free so it's unit-testable without a live GeometryReader
// or drag gesture. Ports the web reference's
// `mui-split-layout.component.ts` decisions: Detail collapses first (the
// most dispensable region — an inspector or thread panel), then Sidebar,
// leaving Center full-width on a phone.

import Foundation

enum MuiSplitLayoutMath {
    /// Below this host width, Detail collapses even if `showDetail` is
    /// true — there's no room for a third column. Matches the web
    /// reference's `DETAIL_COLLAPSE_PX`.
    static let detailCollapsePx: Double = 900
    /// Below this host width, Sidebar collapses too, leaving Center alone.
    /// Matches the web reference's `SIDEBAR_COLLAPSE_PX`.
    static let sidebarCollapsePx: Double = 640

    static func clamp(_ value: Double, min: Double, max: Double) -> Double {
        Swift.min(max, Swift.max(min, value))
    }

    /// Whether the Detail region should render at all: hidden outright when
    /// the caller has nothing to put there (`showDetail == false`), or when
    /// the host is too narrow and collapse is enabled.
    static func detailCollapsed(showDetail: Bool, hostWidth: Double, collapseEnabled: Bool) -> Bool {
        !showDetail || (collapseEnabled && hostWidth < detailCollapsePx)
    }

    /// Whether the Sidebar region should render at all.
    static func sidebarCollapsed(hostWidth: Double, collapseEnabled: Bool) -> Bool {
        collapseEnabled && hostWidth < sidebarCollapsePx
    }
}

File: src/windows/Maple.WinUI/MapleUI/Templates/MuiSplitLayoutMath.cs
using System;

namespace Maple.UI
{
    /// <summary>
    /// Pure clamp/collapse math behind <see cref="MuiSplitLayout"/> —
    /// unit-tested without a live Window. Mirrors
    /// `mui-split-layout.component.ts`'s two-stage collapse: Detail
    /// collapses first (900px), then Sidebar (640px), leaving Center
    /// full-width on a phone.
    /// </summary>
    public static class MuiSplitLayoutMath
    {
        /// <summary>Below this host width, Detail collapses (hidden) even
        /// if <c>showDetail</c> is true — there's no room for a third
        /// column.</summary>
        public const double DetailCollapsePx = 900.0;

        /// <summary>Below this host width, Sidebar collapses too, leaving
        /// Center alone.</summary>
        public const double SidebarCollapsePx = 640.0;

        public static double Clamp(double value, double min, double max) =>
            Math.Min(max, Math.Max(min, value));

        public static bool SidebarCollapsed(double hostWidth, bool collapseEnabled) =>
            collapseEnabled && hostWidth < SidebarCollapsePx;

        public static bool DetailCollapsed(bool showDetail, double hostWidth, bool collapseEnabled) =>
            !showDetail || (collapseEnabled && hostWidth < DetailCollapsePx);
    }
}
